### PR TITLE
FEC-10389 Add support for referenceId filter in provider

### PR DESCRIPTION
**Pull request by chausov**

### CODE REVIEW

File: Sources/OVP/Provider/OVPMediaProvider.swift [hunk 1]
```diff
@@ -136,16 +147,17 @@ public enum OVPMediaProviderError: PKError {
             return
         }
         
-        // entryId is requierd
-        guard let entryId = self.entryId else {
-            PKLog.debug("Proivder must have entryId")
-            callback(nil, OVPMediaProviderError.invalidParam(paramName: "entryId"))
+        // entryId or referenceId requierd
```
Comment (x-NR-x) [inline Sources/OVP/Provider/OVPMediaProvider.swift]: Please update it on the way to 'required' (fix typo)

File: Sources/OVP/Services/OVPBaseEntryService.swift [hunk 2]
```diff
@@ -15,14 +15,19 @@ import KalturaNetKit
...
                                    "type": 1] as [String: Any]
             request.setBody(key: "ks", value: JSON(ks))
                 .setBody(key: "responseProfile", value: JSON(responseProfile))
-                .setBody(key: "filter:redirectFromEntryId", value: JSON(entryID))
+                
+            if let entryID = entryID {
+                request.setBody(key: "filter:redirectFromEntryId", value: JSON(entryID))
+            } else if let referenceId = referenceId {
+                request.setBody(key: "filter:referenceIdEqual", value: JSON(referenceId))
+            }
```
Comment (x-NR-x) [inline Sources/OVP/Services/OVPBaseEntryService.swift:30]: Can there be an entryID and referenceId?

Comment (chausov) [reply Sources/OVP/Services/OVPBaseEntryService.swift:30]: Same logic we have on [Android](https://github.com/kaltura/playkit-android-providers/blob/df96831ee5edfbe23a39b7996c526d6afc12fef8/mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/services/BaseEntryService.java#L61) and [WEB](https://github.com/kaltura/playkit-js-providers/blob/dd68082fb32be21e26cc92439c1d9628b581c8a7/src/k-provider/ovp/services/base-entry-service.js#L68)

File: Sources/OVP/Provider/OVPMediaProvider.swift [hunk 3]
```diff
@@ -136,16 +147,17 @@ public enum OVPMediaProviderError: PKError {
             return
         }
         
-        // entryId is requierd
-        guard let entryId = self.entryId else {
-            PKLog.debug("Proivder must have entryId")
-            callback(nil, OVPMediaProviderError.invalidParam(paramName: "entryId"))
+        // entryId or referenceId required
+        if self.entryId == nil && self.referenceId == nil {
+            PKLog.debug("Proivder must have entryId or referenceId")
```
Comment (x-NR-x) [inline Sources/OVP/Provider/OVPMediaProvider.swift:152]: Please update typo here as well for 'Provider'. Thanks.

**Review by x-NR-x — APPROVED** [review]

**Pull request merged**